openff/qcsubmit/factories.py

```diff
@@ -28,41 +28,49 @@
 from openff.qcsubmit.serializers import deserialize, serialize
 from openff.qcsubmit.workflow_components import CustomWorkflowComponent, get_component
 
+C = TypeVar("C", bound=CustomWorkflowComponent)
+T = TypeVar("T", bound=BasicDataset)
 
-class BasicDatasetFactory(CommonBase):
-    """
-    Basic dataset generator factory used to build work flows using workflow components before executing them to generate
-    a dataset.
 
-    The basic dataset is ideal for large collections of single point calculations using any of the energy, gradient or
-    hessian drivers.
+class BaseDatasetFactory(CommonBase, abc.ABC):
+    """
+    The Base factory which all other dataset factories should inherit from.
     """
 
-    factory_type: Literal["BasicDatasetFactory"] = Field(
-        "BasicDatasetFactory",
+    factory_type: Literal["BaseDatasetFactory"] = Field(
+        "BaseDatasetFactory",
         description="The type of dataset factory which corresponds to the dataset made.",
     )
-    workflow: Dict[str, CustomWorkflowComponent] = Field(
-        {},
+    workflow: List[C] = Field(
```

Comment:
I think for this to work fully in pydantic this needs to be `List[Union[StandardConformerGenerator, MolecularWeightFilter, ..., etc]]` where all possible components are listed here

Reply:
Ah yeah just tested it and it does need to be a full list of components thanks for catching this.